examples/serverless/main.tf

```diff
@@ -0,0 +1,284 @@
+provider "aws" {
+  region = local.region
+}
+
+data "aws_availability_zones" "available" {}
+
+locals {
+  region = "us-east-1"
+  name   = "dms-ex-${basename(path.cwd)}"
+
+  vpc_cidr = "10.0.0.0/16"
+  azs      = slice(data.aws_availability_zones.available.names, 0, 3)
+
+  db_name     = "example"
+  db_username = "example"
+  db_password = "password123!" # do better!
+
+  tags = {
+    Name       = local.name
+    Example    = local.name
+    Repository = "https://github.com/terraform-aws-modules/terraform-aws-dms"
+  }
+}
+
+################################################################################
+# DMS Module
+################################################################################
+
+module "dms_aurora_postgresql_aurora_mysql" {
+  source = "../.."
+
+  # Subnet group
+  repl_subnet_group_name        = local.name
+  repl_subnet_group_description = "DMS Subnet group for ${local.name}"
+  repl_subnet_group_subnet_ids  = module.vpc.private_subnets
+
+  # Instance
+  create_repl_instance = false
+
+  # Access role
+  create_access_iam_role = true
+  access_secret_arns = [
+    module.secrets_manager_postgresql.secret_arn,
+    module.secrets_manager_mysql.secret_arn,
+  ]
+
+  # Endpoints
+  endpoints = {
+    postgresql-source = {
+      database_name               = local.db_name
+      endpoint_id                 = "${local.name}-postgresql-source"
+      endpoint_type               = "source"
+      engine_name                 = "aurora-postgresql"
+      extra_connection_attributes = "heartbeatFrequency=1;secretsManagerEndpointOverride=${module.vpc_endpoints.endpoints["secretsmanager"]["dns_entry"][0]["dns_name"]}"
+      secrets_manager_arn         = module.secrets_manager_postgresql.secret_arn
+
+      postgres_settings = {
+        capture_ddls        = false
+        heartbeat_enable    = true
+        heartbeat_frequency = 1
+      }
+
+      tags = { EndpointType = "postgresql-source" }
+    }
+
+    mysql-destination = {
+      database_name               = local.db_name
+      endpoint_id                 = "${local.name}-mysql-destination"
+      endpoint_type               = "target"
+      engine_name                 = "aurora"
```

Comment:
Shouldn't it be `aurora-mysql` like in the `rds_aurora` module block in `for_each` ?

Reply:
This is actually the `engine_name` used by `aws_dms_endpoint` resource: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/dms_endpoint#engine_name

It confused me as well to be honest, but given its an Aurora RDS instance and not a self-managed MySQL instance this appears to be the correct option, and worked in testing.